resource_customizations/spot.io/SpotDeployment/health.lua

```diff
@@ -0,0 +1,41 @@
+hs = {}
+
+if obj.status == nil or obj.status.conditions == nil then
+  return hs
+end
+
+for _, condition in ipairs(obj.status.conditions) do
+  if condition.type == "InvalidSpec" then
+    hs.status = "Degraded"
+    hs.message = condition.message
+    return hs
+  end
+  if condition.type == "Progressing" and condition.reason == "RolloutAborted" then
+    hs.status = "Degraded"
+    hs.message = condition.message
+    return hs
+  end
+  if condition.type == "Progressing" and condition.reason == "ProgressDeadlineExceeded" then
+    hs.status = "Degraded"
+    hs.message = condition.message
+    return hs
+  end
+  if condition.type == "Paused" and condition.status == "True" then
+    hs.status = "Suspended"
+    hs.message = "Rollout is paused"
+    return hs
+  end
+end
+
+if obj.status.phase == "Progressing" then
+  hs.status = "Progressing"
+  hs.message = "Waiting for rollout to finish steps"
+  return hs
+end
+
```

Comment:
if ` SpotDeployment` is kind of rollout, do you need to add more health checks similar to rollout? - https://github.com/argoproj/argo-cd/blob/dd70eccbcab6a3300bd2df76ea08cc942ec1170b/resource_customizations/argoproj.io/Rollout/health.lua

Reply:
SpotDeployment is not kind of rollout so it not need similar health checks to argo rollout.

Reply:
messages look confusing to me, it says `rollout is paused` .  I don't know spotdeployment spec much. If you think the message is right, you can resolve the comment